### ISSUE TITLE
v0.54 500 error from sql query in custom template; code worked in v0.53; found a workaround

### ISSUE THREAD

**Issue by jrdmb** [simonw/datasette-template-sql#3]
v0.54 500 error in sql query template; code worked in v0.53; found a workaround

**schema:**   
CREATE TABLE "talks" ("talk" TEXT,"series" INTEGER, "talkdate" TEXT)  
CREATE TABLE "series" ("id" INTEGER PRIMARY KEY, "series" TEXT, talks_list TEXT default '', website TEXT default '');

**Live example of correctly rendered template in v.053:** https://cosmotalks-cy6xkkbezq-uw.a.run.app/cosmotalks/talks/1

**Description of problem:** I needed 'sql select' code in a custom row-mydatabase-mytable.html template to lookup the series name for a foreign key integer value in the talks table.  So `metadata.json` specifies the `datasette-template-sql` plugin.

The code below worked perfectly in v0.53 (just the relevant sql statement part is shown; full code is [here](https://github.com/jrdmb/cosmotalks-datasette/blob/main/templates/row-cosmotalks-talks.html)):

```
{# custom addition #}  
{% for row in display_rows %}  
    ...  
    {% set sname = sql("select series from series where id = ?", [row.series]) %}  
    <strong>Series name: {{ sname[0].series }}  
    ...  
{% endfor %}  
{# End of custom addition #}  
```

**In v0.54, that code resulted in a 500 error with a 'no such table series' message.**  A second query in that template also did not work but the above is fully illustrative of the problem.

All templates were up-to-date along with datasette v0.54.

**Workaround:** After fiddling around with trying different things, what worked was the syntax from [Querying a different database from the datasette-template-sql github repo](https://github.com/simonw/datasette-template-sql#querying-a-different-database) to add the database name to the sql statement:

`{% set sname = sql("select series from series where id = ?", [row.series], database="cosmotalks") %}`

Though this was found to work, it should not be necessary to add `database="cosmotalks"` since per the `datasette-template-sql` README, it's only needed when querying a different database, but here it's a table within the same database.

**Comment by simonw** [comment]
Running `pytest` after first running `pip install -U datasette` throws an error:

```
(datasette-template-sql) datasette-template-sql % pytest
========================================================================= test session starts =========================================================================
platform darwin -- Python 3.8.6, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /Users/simon/Dropbox/Development/datasette-template-sql
plugins: asyncio-0.11.0
collected 3 items                                                                                                                                                     

tests/test_template_sql.py .F.                                                                                                                                  [100%]

============================================================================== FAILURES ===============================================================================
__________________________________________________________________ test_sql_against_default_database __________________________________________________________________

app = <datasette.utils.asgi.AsgiLifespan object at 0x110dc7280>

    @pytest.mark.asyncio
    async def test_sql_against_default_database(app):
        async with httpx.AsyncClient(app=app) as client:
            response = await client.get("http://localhost/")
            assert 200 == response.status_code
            stripped = re.sub(r"\s+", " ", response.text)
>           assert "<pre> type: table<br> name: dogs<br> </pre>" in stripped
E           assert '<pre> type: table<br> name: dogs<br> </pre>' in '<!DOCTYPE html> <html> <head> <title>Datasette: dogs, news</title> <link rel="stylesheet" href="/-/static/app.css?4e3...n ).forEach(details => details.open = false); }); </script> <!-- Templates considered: *index.html --> </body> </html>'

tests/test_template_sql.py:46: AssertionError
```

**Comment by simonw** [comment]
Here's why: https://github.com/simonw/datasette-template-sql/blob/c1a4a5038c43c61ebd20029254d088edd7f09356/datasette_template_sql/__init__.py#L6-L8

The `.get_database()` method implements this logic correctly now.

**Comment by simonw** [comment]
Fix is now deployed to https://www.niche-museums.com/ - on my laptop it exhibited the 500 error with `1.0.1` but was fixed with `1.0.2`.

https://www.niche-museums.com/-/plugins

**Comment by jrdmb** [comment]
Just installed `datasette-template-sql 1.0.2` and confirming on my end that the code is now working.

Many thanks!
